src/fsharp/tast.fs

```diff
@@ -2311,13 +2326,19 @@ and [<StructuredFormatDisplay("{LogicalName}")>]
     member x.IsExtensionMember          = x.val_flags.IsExtensionMember
 
     /// The quotation expression associated with a value given the [<ReflectedDefinition>] tag
-    member x.ReflectedDefinition        = x.val_defn
+    member x.ReflectedDefinition        =
+        match x.val_opt_data with
+        | Some x -> x.val_defn
```

Comment:
Don't re-use the name `x` in all of these - just call it `optData` or something :)